client/consul.go

```diff
@@ -100,15 +124,73 @@ func (c *ConsulClient) GetLicense(ctx context.Context, q *consulapi.QueryOptions
 	var license string
 
 	f := func(context.Context) error {
+		var err error
 		license, err = c.Operator().LicenseGetSigned(q)
+
+		// Process the error by wrapping it in the correct error types
 		if err != nil {
-			license = ""
-			return err
+			statusCode := getStatusCodeFromError(ctx, err)
+
+			// If we get a StatusNotFound assume that this is because CTS
+			// is connected to OSS Consul where this endpoint isn't available
+			// wrap in the appropriate error
+			if statusCode == http.StatusNotFound {
+				err = &NonEnterpriseConsulError{Err: err}
+			}
+
+			// non-retryable errors allows for termination of retries
+			if isStatusCodeRetryable(statusCode) {
+				return err
+			} else {
+				return &retry.NonRetryableError{Err: err}
+			}
```

Comment:
To be complete the parity, what about this?

```Go
			// If we get a StatusNotFound assume that this is because CTS
			// is connected to OSS Consul where this endpoint isn't available
			// wrap in the appropriate error
			if statusCode == http.StatusNotFound {
				err = &NonEnterpriseConsulError{Err: err}
			}

			// non-retryable errors allows for termination of retries
			if !isResponseCodeRetryable(statusCode) {
				err = &retry.NonRetryableError{Err: err}
			}

			return err
```

Reply:
Yes exactly ✨